packages/flutter/lib/src/widgets/interactive_viewer.dart

```diff
@@ -492,22 +492,27 @@ class _InteractiveViewerState extends State<InteractiveViewer> with TickerProvid
   // This value was eyeballed to give a feel similar to Google Photos.
   static const double _kDrag = 0.0000135;
 
+  // Cache the size of the child because when it changes, the boundaryRect needs
+  // to be recalculated.
+  Size _childSize;
+
   // The _boundaryRect is calculated by adding the boundaryMargin to the size of
   // the child.
   Rect _boundaryRectCached;
   Rect get _boundaryRect {
-    if (_boundaryRectCached != null) {
-      return _boundaryRectCached;
-    }
     assert(_childKey.currentContext != null);
     assert(!widget.boundaryMargin.left.isNaN);
     assert(!widget.boundaryMargin.right.isNaN);
     assert(!widget.boundaryMargin.top.isNaN);
     assert(!widget.boundaryMargin.bottom.isNaN);
 
     final RenderBox childRenderBox = _childKey.currentContext.findRenderObject() as RenderBox;
-    final Size childSize = childRenderBox.size;
-    _boundaryRectCached = widget.boundaryMargin.inflateRect(Offset.zero & childSize);
+    if (childRenderBox.size == _childSize && _boundaryRectCached != null) {
```

Comment:
nit: at this point the caching does seem to be worth doing anymore? You need to find the child's render box in order to verify the cache is still valid. It's only avoiding redoing the `inflateRect` calculation which seems cheap.

Reply:
Good call.  I removed the caching.  I'm assuming `findRenderObject` shouldn't be too much of a performance hit, but I'll watch out for slowdown.